auth/realm/base/src/main/java/org/wildfly/security/auth/realm/FileSystemSecurityRealm.java

```diff
@@ -676,6 +758,19 @@ private Path tempPath() {
         public void create() throws RealmUnavailableException {
             if (System.getSecurityManager() == null) {
                 createPrivileged();
+                if(privateKey != null) {
+                    try {
+                        createDigitalSignature(path.toString(), this.name);
+                        writeToMainIndex(this.mainIndex, this.path, this.secretKey);
```

Comment:
These write operations should be moved to `createPrivileged()`.

Reply:
Looks like `createDigitalSignature` can throw a `RealmUnavailableException`. Since that extends `IOException`, we'd get to `ElytronMessages.log.unableToAccessMainIndex` in this case too.